From 9e3c7c7ba58b0b3ca1f6da8d9336eb859fb853fa Mon Sep 17 00:00:00 2001
From: Yao Xiao <xiaoyao@rock-chips.com>
Date: Tue, 31 Jul 2018 09:53:55 +0800
Subject: [PATCH] net: wireless: fix wifi disconnect when suspended

Change-Id: Ifbe36bc9cbfd12a7cd206ab0828cfda20af43b6a
Signed-off-by: Yao Xiao <xiaoyao@rock-chips.com>
(cherry picked from commit 9eb169172a8140fd6a0bc2dc79a21c4f7ce0185b (develop-3.10))
---
 net/wireless/sysfs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/net/wireless/sysfs.c b/net/wireless/sysfs.c
index 9cee0220665d..86440be2bfde 100644
--- a/net/wireless/sysfs.c
+++ b/net/wireless/sysfs.c
@@ -104,8 +104,8 @@ static int wiphy_suspend(struct device *dev)
 
 	rtnl_lock();
 	if (rdev->wiphy.registered) {
-		if (!rdev->wiphy.wowlan_config)
-			cfg80211_leave_all(rdev);
+		//if (!rdev->wiphy.wowlan_config)
+		//	cfg80211_leave_all(rdev);
 		if (rdev->ops->suspend)
 			ret = rdev_suspend(rdev, rdev->wiphy.wowlan_config);
 		if (ret == 1) {
-- 
2.35.3

